app/va/va_profile/va_profile_client.py

```diff
@@ -115,21 +127,27 @@ def get_profile(self, va_profile_id: RecipientIdentifier) -> Profile:
                 )
             response.raise_for_status()
         except (requests.HTTPError, requests.RequestException, requests.Timeout) as e:
-            self._handle_exceptions(va_profile_id.id_value, e)
+            self._handle_exceptions(initial_id, e)
 
         response_json: dict = response.json()
         return response_json.get('profile', {})
 
-    def get_mobile_telephone_from_contact_info(self, contact_info: ContactInformation) -> Optional[str]:
+    def get_mobile_telephone_from_contact_info(self, contact_info: ContactInformation) -> str | None:
         """
         Find the most recently created mobile phone number from a veteran's Vet360 contact information
 
         Args:
-            contact_info (ContactInformation):  Contact Information object retrieved from Vet360 API endpoint
+            contact_info (ContactInformation):  Unencrypted contact information retrieved from Vet360 API endpoint
 
         Returns:
             string representation of the most recently created mobile phone number, or None
         """
+
+        va_profile_id: int | None = contact_info.get('vaProfileId')
+        if va_profile_id is not None and is_feature_enabled(FeatureFlag.PII_ENABLED):
+            # Encrypt the value, which might be logged.
+            va_profile_id = PiiVaProfileID(va_profile_id).get_encrypted_value()
```

Comment:
We don't need to specify what value we need when assigning it here. This may be confusing in the future since it's mainly being used for logging. In logs I'd rather see the string interpretation of the object since it will either be the value (if set to low), or `redacted IdType` if set to something else. It should use the PiiVaProfileID.\_\_str\_\_() value in the logs right? If not you can cast it in the log messages.

```suggestion
        va_profile_id: int | None = contact_info.get('vaProfileId')
        if va_profile_id is not None and is_feature_enabled(FeatureFlag.PII_ENABLED):
            # Encrypt the value, which might be logged.
            va_profile_id = PiiVaProfileID(va_profile_id)
```

Reply:
changed